### PR TITLE
Added env GUNICORN_WORKER to open up for other worker_class

### PR DESCRIPTION
**Pull request by jameswnl**
To allow using e.g. `gthread` worker_class to prevent long running worker being shutdown

### CODE REVIEW

**Comment by jameswnl** [comment]
@tumido I know `gthread` works for the `Host Inventory` data-collector. Didn't pin it in this PR because I am not sure about the older types of data-collector.

**Comment by tumido** [comment]
@jameswnl, if you want to use it, the `gthread` it should be available on the environment. Is `gevent` installed by default or should we add something to the `Pipenv`?

**Comment by jameswnl** [comment]
`gthread` comes with `gunicorn`. no need to install it.

In our case, we don't really care that much about `sync` or `async` worker types because we always spawn a thread to do the long running collection work.

The problem we have now is that `gunicorn` will kill the worker after certain period of time when it can't "talk" to the worker.  It will assume the worker is hanging and so will kill it.  Using `gthread` allows `gunicorn` master process to know the process is not hanging.

File: config.py [hunk 1]
```diff
@@ -4,8 +4,7 @@
 bind = '127.0.0.1:8000'
 
 workers = int(os.environ.get('GUNICORN_PROCESSES', '3'))
-threads = int(os.environ.get('GUNICORN_THREADS', '1'))
```
Comment (jameswnl) [inline config.py]: meaningless for non-sync worker

File: config.py [hunk 2]
```diff
@@ -4,8 +4,7 @@
...
-timeout = 120
```
Comment (jameswnl) [inline config.py]: no need of this anymore.

**Comment by jameswnl** [comment]
@AparnaKarve doesn't think this is necessary and that she already have the `timeout=120` in place and that for instance type cost savings sometimes takes 30-40 minutes for data collection, and that works fine.


We can close this for now.

**Comment by AparnaKarve** [comment]
@jameswnl I tested this with long running data collection jobs in Instance Type validation and Idle Cost savings, and it seems to be working well.

I would love to do more testing on this, especially on those corner case scenarios like when the Topology Inventory Service is overloaded, because of which it sometimes could take up to 45+ minutes to complete the data collection.
I guess the best way to test this would be to merge and evaluate on a day-to-day basis.

Hence merging this so we can observe and evaluate.
Thanks!
